### ISSUE TITLE
dojo> (reap %) results in bail: fail bailing out

### ISSUE THREAD

**Issue by mikolajpp** [urbit/arvo#519]
Executing the following line:
`(reap %)`

triggers bail. 
Urbit is not damaged, and can boot again. 

Reproducible every time.

**Comment by ohAitch** [comment]
(reap %) is (reap '~zod' /home/~2017....), which is approximately (reap
50.000 %constant). If the error is bail: meme, then your urbit is running
out of memory; which probably shouldn't be treated as a fatal error, but at
any rate indeed causes no lasting harm.

**Comment by mikolajpp** [comment]
Nope, it fails with
```
bail: fail
bailing out
zsh: abort (core dumped)
```

**Comment by ohAitch** [comment]
Hm, maybe the text of the ship is >31 bits, and the jet crashes because it
doesn't like the indirect atom representation(and can predict what happens
when you try to cons together >31 bits in unary)

**Comment by mikolajpp** [comment]
The ship in question is a planet: `~tonweb-pilseb`

**Comment by ohAitch** [comment]
Yeah, probably that then; or possibly an error elsewhere, the one thing I'm
confident in is using text as a repitition count is in fact an inadvised
operation

Did you perhaps mean (ream '%'), or ??? %

**Comment by mikolajpp** [comment]
I just randomly typed that and it crashed ;-)
All things equal, it still should nest.fail or something?
Or is bail: fail acceptable operation?

**Comment by ohAitch** [comment]
It should probably nest fail yeah, reap should take @u not @ as the counter

It's not super great behavior yeah, I'd file that under a general
urbit/urbit "jet exits bail instead of running the native code" issue.
We're not focusing on sandboxing untrusted code right now, but we'd
definitely like to eventually…

**Comment by belisarius222** [comment]
I'm not sure it's a good idea to run the hoon version of a function whenever its jet errors. It seems dangerous. Changing `++reap` to take `@u` sounds reasonable, though.

**Comment by jtobin** [comment]
The OP example `bail: meme`'s now, and I've updated the issue title to more accurately capture the issue itself.
